Ff111 implements workers modules

hamishwillee opened the pull request:
FF111 supports module loading from workers in https://bugzilla.mozilla.org/show_bug.cgi?id=1247687
This updates the existing support statements. 

Other docs work for this can be tracked in https://github.com/mdn/content/issues/24402

lgarron commented:
I'm glad to see Firefox getting close, but Firefox 111 still **doesn't support dynamic import in module workers**, which means that many apps will still not work.

I wouldn't hardline insist that dynamic import is "part" of module worker support, as much as the former needs the latter to realize its full potential. But I'd go so far as to say that most web devs would expect them together, and module worker support without dynamic import is **more likely* to break apps unexpectedly (compared to no support for module workers), as this has been a safe way to implement feature detection so far:

- Try to instantiate a module worker.
  - If unsuccessful, show a useful error or a fallback.
  - If successful, assume that it can use dynamic imports.

Due to worker quirks, you don't automatically the lack of dynamic import support in a worker as an error either, unless you implement manual support to do so (e.g. using `postMessage`).

For web devs looking up compatibility, would it be possible to mark this as partial support, or at least include a note?

hamishwillee commented:
Absolutely. I'll confirm this with Firefox dev team, but if it does not support dynamic import that needs to be noted (in some way).

evilpie commented:
It's not shipping by default in Firefox 111, also I suspect because dynamic imports is missing.
Put behind Nightly-only pref here: https://bugzilla.mozilla.org/show_bug.cgi?id=1812628
Bug for enabling modules by default: https://bugzilla.mozilla.org/show_bug.cgi?id=1812591

hamishwillee commented:
Thanks very much @evilpie. I've updated with a flagged version and "preview" to indicate that this is available in nightly. Also updated `import`. I've suggested #19033 be closed since this is more correct and complete (now).

I've separately updated `import()` in  #19054 to indicate dynamic module import is not supported in workers - would appreciate your guidance on whether there is a WPT test I can use to verify the data?

@queengooborg @Elchi3 This should be correct now.
